pytorch_lightning/__init__.py

```diff
@@ -10,9 +10,10 @@
 __docs__ = "PyTorch Lightning is the lightweight PyTorch wrapper for ML researchers." \
            " Scale your models. Write less boilerplate."
 
-from logging import getLogger
+import logging as python_logging
 
-_logger = getLogger("lightning")
+_logger = python_logging.getLogger("lightning")
+python_logging.basicConfig(level=python_logging.INFO)
```

Comment:
```
from logging import getLogger, INFO as lvl_INFO
_logger = getLogger()
_logger.setLevel(lvl_INFO)
```

Reply:
this doesn't set up the stream handler to print to stdout

Reply:
So you tried it?
The problem with `basisConfig` is that if the logging is called earlier this doesn't help either...

Reply:
just check, my proposal solves the problem above, this solution does not...
you can simply check it on your own test adding on line 11
```
import logging
logging.info("sdwsd")
```

Reply:
your proposed changes don't resolve the root issue of configuring logging to be displayed by default. here's a quick test:

using 
```
import logging as python_logging
_logger = python_logging.getLogger("lightning")
python_logging.basicConfig(level=python_logging.INFO)
```

we get
```
>>> from pl_examples.basic_examples.lightning_module_template import LightningTemplateModel
/Users/jeremyjordan/anaconda/envs/lightning/lib/python3.7/site-packages/graphql/type/directives.py:62: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
  assert isinstance(args, collections.Mapping), '{} args must be a dict with argument names as keys.'.format(name)
/Users/jeremyjordan/anaconda/envs/lightning/lib/python3.7/site-packages/graphql/type/typemap.py:1: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
  from collections import OrderedDict, Sequence, defaultdict
>>> from pytorch_lightning import Trainer
>>> from argparse import Namespace
>>> 
>>> hparams = {"batch_size": 32,"in_features": 784,"hidden_dim": 128,"drop_prob": 0.2,"out_features": 10,"learning_rate": 1e-3,"data_root": "data"}
>>> hparams = Namespace(**hparams)
>>> model = LightningTemplateModel(hparams)
>>> trainer = Trainer(profiler=True, max_epochs=1)
/Users/jeremyjordan/Documents/Code/pytorch-lightning/pytorch_lightning/callbacks/gradient_accumulation_scheduler.py:45: DeprecationWarning: Epochs indexing of `scheduling` starts from "1" until v0.6.x, but will start from "0" in v0.8.0.
  ' but will start from "0" in v0.8.0.', DeprecationWarning)
>>> trainer.fit(model)
INFO:lightning:
  | Name      | Type        | Params | In sizes | Out sizes
-------------------------------------------------------------
0 | c_d1      | Linear      | 100 K  | [5, 784] | [5, 128] 
1 | c_d1_bn   | BatchNorm1d | 256    | [5, 128] | [5, 128] 
2 | c_d1_drop | Dropout     | 0      | [5, 128] | [5, 128] 
3 | c_d2      | Linear      | 1 K    | [5, 128] | [5, 10]  
INFO:lightning:Validation data loader called.
0it [00:00, ?it/s]/Users/jeremyjordan/Documents/Code/pytorch-lightning/pytorch_lightning/trainer/training_loop.py:285: RuntimeWarning: Displayed epoch numbers in the progress bar start from "1" until v0.6.x, but will start from "0" in v0.8.0.
  ' but will start from "0" in v0.8.0.', RuntimeWarning)
INFO:lightning:Training data loader called.
INFO:lightning:Validation data loader called.
Epoch 1: 100%|█████████████████████████████████████████████| 2188/2188 [00:32<00:00, 68.07it/s, loss=0.146, train_loss=0.0795, v_num=34, val_acc=0.937, val_loss=0.214]
INFO:lightning:                                                                                                                                                        

Profiler Report

Action                  |  Mean duration (s)    |  Total time (s) 
-----------------------------------------------------------------
on_train_start          |  1.1643e-05           |  1.1643e-05     
on_epoch_start          |  3.5056e-05           |  3.5056e-05     
get_train_batch         |  0.0091583            |  17.181         
on_batch_start          |  1.8774e-05           |  0.035202       
model_forward           |  0.0018705            |  3.5072         
model_backward          |  0.0019597            |  3.6745         
on_after_backward       |  4.2778e-06           |  0.0080209      
optimizer_step          |  0.0010669            |  2.0004         
on_batch_end            |  1.9615e-05           |  0.036779       
on_epoch_end            |  1.5511e-05           |  1.5511e-05     
on_train_end            |  1.4772e-05           |  1.4772e-05     

1
```

---

using as you suggest
``` 
from logging import getLogger, INFO as lvl_INFO
_logger = getLogger()
_logger.setLevel(lvl_INFO)
```

we get 

```
>>> from pl_examples.basic_examples.lightning_module_template import LightningTemplateModel
/Users/jeremyjordan/anaconda/envs/lightning/lib/python3.7/site-packages/graphql/type/directives.py:62: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
  assert isinstance(args, collections.Mapping), '{} args must be a dict with argument names as keys.'.format(name)
/Users/jeremyjordan/anaconda/envs/lightning/lib/python3.7/site-packages/graphql/type/typemap.py:1: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working
  from collections import OrderedDict, Sequence, defaultdict
>>> from pytorch_lightning import Trainer
>>> from argparse import Namespace
>>> 
>>> hparams = {"batch_size": 32,"in_features": 784,"hidden_dim": 128,"drop_prob": 0.2,"out_features": 10,"learning_rate": 1e-3,"data_root": "data"}
>>> hparams = Namespace(**hparams)
>>> model = LightningTemplateModel(hparams)
>>> trainer = Trainer(profiler=True, max_epochs=1)
/Users/jeremyjordan/Documents/Code/pytorch-lightning/pytorch_lightning/callbacks/gradient_accumulation_scheduler.py:45: DeprecationWarning: Epochs indexing of `scheduling` starts from "1" until v0.6.x, but will start from "0" in v0.8.0.
  ' but will start from "0" in v0.8.0.', DeprecationWarning)
>>> trainer.fit(model)
0it [00:00, ?it/s]/Users/jeremyjordan/Documents/Code/pytorch-lightning/pytorch_lightning/trainer/training_loop.py:285: RuntimeWarning: Displayed epoch numbers in the progress bar start from "1" until v0.6.x, but will start from "0" in v0.8.0.
  ' but will start from "0" in v0.8.0.', RuntimeWarning)
Epoch 1: 100%|███████████████████████████████████████████████| 2188/2188 [00:32<00:00, 66.81it/s, loss=0.152, train_loss=0.04, v_num=35, val_acc=0.934, val_loss=0.217]
1         
```

Reply:
> The problem with basisConfig is that if the logging is called earlier this doesn't help either...

but since we call this in the root init file, the basicConfig is always going to be called before Lightning code execute since it is triggered on importing Lightning

imo `basicConfig` is exactly what we want, it sets up default stream handlers unless the root logger has already set up logging. this means that if the user does nothing, they'll see our logs. if they want more control over what is logged, our code doesn't get in the way of that. 

from the [docs](https://docs.python.org/3/library/logging.html#logging.basicConfig)
> This function does nothing if the root logger already has handlers configured, unless the keyword argument force is set to True.